staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -179,42 +185,50 @@ type crdInfo struct {
 	// update issued when the crdInfo was created. The API server uses the
 	// information to decide whether a CR write request should be allowed,
 	// rejected, or blocked.
-	storageVersionUpdate *storageVersionUpdate
+	storageVersionUpdate *storageVersionUpdateInfo
 }
 
 // crdStorageMap goes from customresourcedefinition to its storage
 type crdStorageMap map[types.UID]*crdInfo
 
-// storageVersionUpdate holds information about a storage version update,
+// storageVersionUpdateInfo holds information about a storage version update,
 // indicating whether the update gets processed, or timed-out.
-type storageVersionUpdate struct {
+type storageVersionUpdateInfo struct {
+	crd *apiextensionsv1.CustomResourceDefinition
+
+	updateChannels *storageVersionUpdateChannels
+
+	// timeout is the time when the API server will unblock and allow CR
+	// write requests even if the storage version update hasn't been
+	// processed.
+	timeout time.Time
```

Comment:
I thought we won't serve CR requests if the SV update fails. WDYT?

Reply:
Oh yes that's what the code is doing, i.e. failing the CR if timeout occurs. Updated the comment to reflect the same.